### ISSUE TITLE
Personal channel name for People chats

### ISSUE THREAD

**Issue by DreamFlasher** [vector-im/riot-web#3130]
Currently, if the channel with another person is not named, than their nickname is shown. 
Problem: 
1) It's impossible with many people to remember everyones nickname.
2) Also it's impossible to convince everyone to use their full real name (and also undesired).
3) Naming the channel in a meaningful way has the problem that it's shown for the other person, too. So currently we use titles for chats where both names are in. 

Possible solutions:
- Make the channel name for People chats a personal/local/account setting (instead of a server? setting)
- Have something like an address book where one can give names for people and display these names when the channel is unnamed

**Comment by uhoreg** [comment]
It would be helpful for users to be able to set a custom name for rooms in general, with the custom name only being shown to the user who set it.  For example, it would be useful if someone is subscribed to both #debian on Freenode and #debian on OFTC to set custom names so that they're distinguishable.

**Comment by DreamFlasher** [comment]
Good point, I didn't think of it, yes actually channel names should personalizable in general.

**Comment by ara4n** [comment]
This is basically https://github.com/matrix-org/matrix-doc/issues/690.  Agreed it'd be nice to have - it's very annoying when someone renames a 1:1 to be called something unhelpful like "Matthew"

**Comment by uhoreg** [comment]
Would also be good to be able to set a personal avatar for the room without affecting other users.

**Comment by turt2live** [comment]
Pet avatars are tracked here ftr: https://github.com/vector-im/riot-web/issues/363

**Comment by MurzNN** [comment]
For don't create new entities in storage, can we store 'pet name' for rooms in `account_data` like `ignored_users` stored now? 
Something like:
```
{
    "content": {
        "room_names": {
            "!QtykxKocfZaZOUrTwp:matrix.org": {
                "name": "Matrix flame"
            }
        }
    },
    "type": "m.room_name_override"
}
```

**Comment by t3chguy** [comment]
Why not just store it in room account data then you don't need the room id inside the event.

**Comment by MurzNN** [comment]
> Why not just store it in room account data then you don't need the room id inside the event.

Does room account data have private storage for each user, that not visible to other room members?

**Comment by t3chguy** [comment]
Yup, that's why it's room *account* data.

**Comment by kiliankoe** [comment]
This would be fantastic to have. I'm one of those people who have trouble remembering usernames of my contacts. I also have groups with certain users that I would love to just label with something related to these users that doesn't make sense for them (e.g. "parents").

**Comment by eternalbit** [comment]
First of all, sorry for not adding real content to the discussion.
Moderators, feel free to delete my comment.

My point: I don't understand why this issue has **p2** assigned. I've literally seen roughly 10 more or less duplicate issues with this same request from users, who also want this feature - going back to 2017.

And then there are issues with **p1** assigned where I really don't get how it's supposed to be more important than this one here ... example: [Why doesn't 'da<tab>' autocomplete Dave in #riot-dev?](https://github.com/vector-im/riot-web/issues/8957)

Sure, priorities are somehow a matter of personal perspective ... but for me, the same issue popping up again and again like this one clearly indicates it's quite important.

And I bet that this is one of the missing features that keep potential users (like me) from using riot/vector.im. I don't want to have to remember silly pseudonyms - which is what many private users chose when making an account. Sure, users from french administration probably use their actual names anyway. But if riot/vector.im becomes more widespreach they won't be a big percentage of all users anymore.

**Comment by dukelec** [comment]
+1

**Comment by MurzNN** [comment]
Seems this is planned to implement in https://github.com/vector-im/roadmap/issues/10 roadmap item?

**Comment by MurzNN** [comment]
I have created the https://github.com/matrix-org/matrix-doc/pull/3015 for this issue.

**Comment by vmario89** [comment]
found that issue because i was looking for the same missing feature. happy to see if we could rename persons locally wihtout overriding global room names, to have a better overview of a whole community and a lot of group members and friends.

**Comment by SZemanek** [comment]
It would be super cool if this could integrate with contacts on the android version of element.

**Comment by ArtexJay** [comment]
Hopefully allowing us to give nicknames or pet names to friends on the app to better distinguish them from other friends will arrive soon.

**Comment by kittykat** [comment]
Moving this issue to discussions in Element meta as we need to make a cross platform decision on how to proceed :+1:
